src/shared-components/button/index.tsx

```diff
@@ -45,7 +45,9 @@ type ButtonProps = {
    */
   isLoading?: boolean;
   loading?: boolean;
-  onClick: () => void;
+  onClick?: (
```

Comment:
Some of our codes uses buttons with `type="submit"`, and then attaches an `onSubmit` handler to the form component. As such, I was incorrect to suggest we should make this required. Sorry!